test/Microsoft.Windows.CsWin32.Tests/GeneratorTests.cs

```diff
@@ -2347,6 +2369,16 @@ private async Task<CSharpCompilation> CreateCompilationAsync(ReferenceAssemblies
             references: metadataReferences,
             options: new CSharpCompilationOptions(OutputKind.DynamicallyLinkedLibrary, platform: platform, allowUnsafe: true));
 
+        this.parseOptions = this.parseOptions.WithPreprocessorSymbols(references.TargetFramework switch
+        {
+            "net20" => ImmutableArray.Create("NETFRAMEWORK", "NET20", "NET20_OR_GREATER"),
+            "net35" => ImmutableArray.Create("NETFRAMEWORK", "NET35", "NET35_OR_GREATER", "NET30_OR_GREATER", "NET20_OR_GREATER"),
+            "net40" => ImmutableArray.Create("NETFRAMEWORK", "NET40", "NET40_OR_GREATER", "NET35_OR_GREATER", "NET30_OR_GREATER", "NET20_OR_GREATER"),
+            "netstandard2.0" => ImmutableArray.Create("NETSTANDARD", "NETSTANDARD2_0", "NETSTANDARD2_0_OR_GREATER", "NETSTANDARD1_6_OR_GREATER", "NETSTANDARD1_5_OR_GREATER", "NETSTANDARD1_4_OR_GREATER", "NETSTANDARD1_3_OR_GREATER", "NETSTANDARD1_2_OR_GREATER", "NETSTANDARD1_1_OR_GREATER", "NETSTANDARD1_0_OR_GREATER"),
+            "net5.0" => ImmutableArray.Create("NET", "NETCOREAPP", "NET5_0", "NET5_0_OR_GREATER", "NETCOREAPP3_1_OR_GREATER", "NETCOREAPP3_0_OR_GREATER", "NETCOREAPP2_2_OR_GREATER", "NETCOREAPP2_1_OR_GREATER", "NETCOREAPP2_0_OR_GREATER", "NETCOREAPP1_1_OR_GREATER", "NETCOREAPP1_0_OR_GREATER"),
+            _ => throw new NotImplementedException("Set up SDK-defined preprocessor symbols for " + references.TargetFramework),
+        });
```

Comment:
I wonder if https://github.com/dotnet/roslyn-sdk/tree/main/src/Microsoft.CodeAnalysis.Testing/Microsoft.CodeAnalysis.Analyzer.Testing would be willing to take over this responsibility (or already has something for this).

Reply:
Oh my, I didn't realize legacy projects were in the scope of support. I'm so used to there being some reason or other that e.g. source generators might not even work for them in the first place. But okay. :+1:

Reply:
Oh yes, certainly. CsWin32 works based on a modern compiler. Old project file formats don't keep it from working. And some of the most p/invoke rich projects that may benefit from CsWin32 are in old codebases.